File: go/hamming/hamming.go
package hamming

import "errors"

var errDifferentLength = errors.New("strands must have the same length")

// Distance calculates the Hamming Distance between two DNA strands
func Distance(a, b string) (int, error) {
	var distance int
	runeA, runeB := []rune(a), []rune(b)

	if len(runeA) != len(runeB) {
		return distance, errDifferentLength
	}

	for i := 0; i < len(runeA); i++ {
		if runeA[i] != runeB[i] {
			distance++
		}
	}

	return distance, nil
}

File: php/hamming/Hamming.php
<?php

declare(strict_types=1);

function distance(string $strandA, string $strandB): int
{
  if(strlen($strandA) != strlen($strandB)) {
    throw new InvalidArgumentException('DNA strands must be of equal length.');
  }

  $diff = 0;

  for ($i = 0; $i < strlen($strandA); $i++) {
    if ($strandA[$i] != $strandB[$i]) {
      $diff++;
    }
  }

  return $diff;
}
